gatsby-node.js

```diff
@@ -306,3 +306,15 @@ exports.onPostBuild = async ({ graphql }) => {
 
   await writeFile('./public/meta.json', JSON.stringify({ allJobIds }));
 };
+
+exports.onCreatePage = async ({
```

Comment:
This seems like a bit of a hack - if we make the blog 404 page a template instead of a page and include it in the build using `createPage` like we do many times above, I believe we can give it a proper `matchPath` straight away and don't have to hack around like this?

Reply:
I followed the documentation example for this kind of situation. You can see it (here)[https://www.gatsbyjs.org/docs/creating-prefixed-404-pages-for-different-languages/]

Reply:
Hmm, my point still stands - for this use case it seems much cleaner to just create the page once from a template, instead of hooking into its creation, where you then delete it and re-create it differently. But if you want to keep it this way, let's at least put the source of this snippet into a comment I'd say